### PR TITLE
Flexible Sync Concept Content: Incorporate tech review feedback

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

This PR incorporates tech review comments from the already-merged #1515 

### Staged Changes (Requires MongoDB Corp SSO)

- [Sync Mode](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fs-concept-tech-review/sync/data-access-patterns/sync-mode/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sync/data-access-patterns/sync-mode.txt [hunk 1]
```diff
@@ -162,3 +197,36 @@ Choose Flexible Sync when:
...
+   have access to data in the ``Billing`` collection. The SDK query 
+   subscription doesn't need any knowledge of this; Flexible Sync handles
+   it automatically by applying the permissions to data that matches the 
+   query.
+
+   .. code-block:: javascript
+
+      const patientVisits = realm.objects("Visit");
+      const patientAndTimeWindow = patientVisits.filtered(
+        'id == "507f191e810c19729de860ea" && procedureDate > "Wed Oct 18 2020 12:00:00 GMT+0000 (UTC)"'
```
Comment (mongomoe) [inline source/sync/data-access-patterns/sync-mode.txt:224]: You can't compare a date to a string, but you can do:
```
procedureDate > new Date("Wed Oct 18 2020 12:00:00 GMT+0000 (UTC)");
```

File: source/sync/data-access-patterns/sync-mode.txt [hunk 2]
```diff
@@ -66,30 +67,64 @@ Choose Partition-Based Sync when:
...
+that a user can access to a synced ``.realm`` file. Each object in a synced 
+realm gets translated to a corresponding document in {+atlas+}.
 
 Choose Flexible Sync when:
 
-- You need to create overlapping query and result sets
+- You need to create expressive query and result sets using a broad range 
+  of queries on up to 10 fields
 - You want to apply permissions at the document level
 - You need a flexible permissioning system to apply more granular permissions
```
Comment (mongomoe) [inline source/sync/data-access-patterns/sync-mode.txt:129]:
```suggestion
- You need a flexible permissions system to apply more granular permissions
```

**Review by mongomoe — APPROVED** [review]
LGTM w/ a couple comments

File: source/sync/data-access-patterns/sync-mode.txt (hunk 2 above)
Comment (dacharyc) [reply source/sync/data-access-patterns/sync-mode.txt:129]: I think we discussed this on the first PR - I actually mean `permissioning` system here, not `permissions`. Good check, though, as I know some grammar checkers will flag it. 👍

**Pull request merged**